packages/libs/src/sdk/api/generator/templates/typescript-fetch/modelGeneric.mustache

```diff
@@ -23,11 +23,11 @@ import {
 /**
  * Check if a given object implements the {{classname}} interface.
  */
-export function instanceOf{{classname}}(value: object): boolean {
+export function instanceOf{{classname}}(value: object): value is {{classname}} {
     let isInstance = true;
     {{#vars}}
     {{#required}}
-    isInstance = isInstance && "{{name}}" in value;
+    isInstance = isInstance && "{{name}}" in value && value["{{name}}"] !== undefined;
```

Comment:
1. Main intent is to fix OneOf models, so that they return "marshalled" correctly. I suppose you could use this for any arbitrary object though

2. The `FromJSONTyped` methods of each model return `undefined` for fields that don't exist on the original response body. The "right" way to do this would be to not have those keys at all, then the previous version of this method would have worked fine, but since they keys exist and are undefined it doesn't. I assume if the key exists but is null it _is_ the type but the field is explicitly nullable.